packages/contracts/README.md

```diff
@@ -27,3 +27,31 @@ interface IFinancialProtocol {
 ```
 
 This protocol allows a service provider to know how long an application has enough funds to run. Of course it is a choice of the service provider off-chain node manager to decide which smart contract implementation to watch for `IFinancialProtocol` events.
+
+## Implementation
+
+The two protocols described above are implemented by two concrete Smart Contracts: `ReaderNodeProvider` and `ValidatorNodeProvider`.
+
+![Alt text](image.png)
+
+### ReaderNodeProvider
+
+The `ReaderNodeProvider` contract is responsible for the implementation of the `IMachineProtocol` protocol and inherits from a `Vault` contract which implements the `IFinancialProtocol` protocol.
+
+Anyone can call the `register` method of the `ReaderNodeProvider` contract to register a new application. The caller must provide the Cartesi machine snapshot location so the service provider can download it and run a node for the application. If the service provider can't find the machine, or if the machine's hash is not the one specified during the application deployment, the node will fail to run.
```

Comment:
The word "method" is never used in the Solidity documentation when referring to smart contract functions, like `register`.